ports/bgfx/vcpkg.json

```
@@ -1,12 +1,11 @@
 {
   "name": "bgfx",
-  "version": "1.127.8725-469",
-  "port-version": 1,
+  "version": "1.128.8777.475",
```

The original version number is `v1.128.8777-475`, see https://github.com/bkaradzic/bgfx.cmake/releases/tag/v1.128.8777-475

```suggestion
  "version": "1.128.8777-475",
```

---

@WangWeiLin-MV It is clear how it is spelled upstream. But is clear that vcpkg correctly compares `1.128.8777-475` and `1.128.8777-476`?

---

I made this change to be more canonical with what `"version"` expects.
I can reverse it.
Can you let me know which way is best?

---

> But is clear that vcpkg correctly compares `1.128.8777-475` and `1.128.8777-476`?

I haven't researched the corresponding code yet, but through a simple test seems to indicate that `vcpkg` correctly compares the [relaxed, dot-separated-, semver-like scheme](https://learn.microsoft.com/en-us/vcpkg/users/versioning#:~:text=relaxed%2C%20dot%2Dseparated%2D%2C%20semver%2Dlike%20scheme)

```pwsh
$_GenPort = {
    param($Ver)
    New-Item -Path ./ports/1port -ItemType Directory -Force | Out-Null
    "{{`n  `"name`": `"1port`",`n  `"version`": `"{0}`"`n}}`n" -f $Ver |
    Out-File -NoNewline ./ports/1port/vcpkg.json
}

$_GenCommit = {
    param($Comment)
    git add .
    git commit -m $Comment --quiet
    vcpkg x-add-version 1port | Out-Null
    git add .
    git commit --amend --no-edit --quiet
}

& $_GenPort -Ver '0.0.0'
& $_GenCommit -Comment "Add 1port 0.0.0"
Write-Host (git rev-parse HEAD)

'1.128.8777',
'1.128.8777-1',
'1.128.8777-1.0',
'1.128.8777-1.1',
'1.128.8777-1.1+a',
'1.128.8777-1.1+b',
'1.128.8777-1.1+bb',
'1.128.8777-2',
'1.128.8777-10',
'1.128.8777-20',
'1.128.8777-a',
'2.0.0' |
ForEach-Object {
    & $_GenPort -Ver $_
    & $_GenCommit -Comment "Add 1port $_"
}
```

```json
{
"dependencies":[


{"name":"1port","version>=":"0.0.0"}

,{"name":"1port","version>=":"1.128.8777"}
,{"name":"1port","version>=":"1.128.8777-1"}
,{"name":"1port","version>=":"1.128.8777-1.0"}

,{"name":"1port","version>=":"1.128.8777-1.1"}
,{"name":"1port","version>=":"1.128.8777-1.1+bb"}
,{"name":"1port","version>=":"1.128.8777-1.1+b"}
,{"name":"1port","version>=":"1.128.8777-1.1+a"}
,{"name":"1port","version>=":"1.128.8777-1.1+b"}
,{"name":"1port","version>=":"1.128.8777-1.1+bb"}
,{"name":"1port","version>=":"1.128.8777-1.1"}

,{"name":"1port","version>=":"1.128.8777-2"}
,{"name":"1port","version>=":"1.128.8777-10"}
,{"name":"1port","version>=":"1.128.8777-20"}
,{"name":"1port","version>=":"1.128.8777-a"}
,{"name":"1port","version>=":"2.0.0"}

],
"builtin-baseline":"---- the commit of the first version 0.0.0 ----"
}
```

Then the command `vcpkg install --dry-run` will find [the lowest version that matches all constraints](https://learn.microsoft.com/en-us/vcpkg/users/versioning#:~:text=the%20lowest%20version%20that%20matches%20all%20constraints), this also means the selected version is the higher precedence between constraints. Then delete the line and repeat.

Got the result of the test, for versioning schema `version`:

`2.0.0` >
`1.128.8777` >
`1.128.8777-a` >
`1.128.8777-20` >
`1.128.8777-10` >
`1.128.8777-2` >
`1.128.8777-1.1` =
`1.128.8777-1.1+a` =
`1.128.8777-1.1+b` =
`1.128.8777-1.1+bb` >
`1.128.8777-1.0` >
`1.128.8777-1` >
`1.1` >
`1.0` >
`1` >
`0.0.0`

---

> But is clear that vcpkg correctly compares `1.128.8777-475` and `1.128.8777-476`?

I haven't researched the corresponding code yet, but through a simple test seems to indicate that `vcpkg` correctly compares the [relaxed, dot-separated-, semver-like scheme](https://learn.microsoft.com/en-us/vcpkg/users/versioning#:~:text=relaxed%2C%20dot%2Dseparated%2D%2C%20semver%2Dlike%20scheme)

```pwsh
$_GenPort = {
    param($Ver)
    New-Item -Path ./ports/1port -ItemType Directory -Force | Out-Null
    "{{`n  `"name`": `"1port`",`n  `"version`": `"{0}`"`n}}`n" -f $Ver |
    Out-File -NoNewline ./ports/1port/vcpkg.json
}

$_GenCommit = {
    param($Comment)
    git add .
    git commit -m $Comment --quiet
    vcpkg x-add-version 1port | Out-Null
    git add .
    git commit --amend --no-edit --quiet
}

& $_GenPort -Ver '0.0.0'
& $_GenCommit -Comment "Add 1port 0.0.0"
Write-Host (git rev-parse HEAD)
'1',
'1.0',
'1.1',
'1.128.8777',
'1.128.8777-1',
'1.128.8777-1.0',
'1.128.8777-1.1',
'1.128.8777-1.1+a',
'1.128.8777-1.1+b',
'1.128.8777-1.1+bb',
'1.128.8777-2',
'1.128.8777-10',
'1.128.8777-20',
'1.128.8777-a',
'2.0.0' |
ForEach-Object {
    & $_GenPort -Ver $_
    & $_GenCommit -Comment "Add 1port $_"
}
```

```json
{
"dependencies":[

{"name":"1port","version>=":"0.0.0"}
,{"name":"1port","version>=":"1"}
,{"name":"1port","version>=":"1.0"}
,{"name":"1port","version>=":"1.1"}

,{"name":"1port","version>=":"1.128.8777"}
,{"name":"1port","version>=":"1.128.8777-1"}
,{"name":"1port","version>=":"1.128.8777-1.0"}

,{"name":"1port","version>=":"1.128.8777-1.1"}
,{"name":"1port","version>=":"1.128.8777-1.1+bb"}
,{"name":"1port","version>=":"1.128.8777-1.1+b"}
,{"name":"1port","version>=":"1.128.8777-1.1+a"}
,{"name":"1port","version>=":"1.128.8777-1.1+b"}
,{"name":"1port","version>=":"1.128.8777-1.1+bb"}
,{"name":"1port","version>=":"1.128.8777-1.1"}

,{"name":"1port","version>=":"1.128.8777-2"}
,{"name":"1port","version>=":"1.128.8777-10"}
,{"name":"1port","version>=":"1.128.8777-20"}
,{"name":"1port","version>=":"1.128.8777-a"}
,{"name":"1port","version>=":"2.0.0"}

],
"builtin-baseline":"---- the commit of the first version 0.0.0 ----"
}
```

Then the command `vcpkg install --dry-run` will find [the lowest version that matches all constraints](https://learn.microsoft.com/en-us/vcpkg/users/versioning#:~:text=the%20lowest%20version%20that%20matches%20all%20constraints), this also means the selected version is the higher precedence between constraints. Then delete the line and repeat.

Got the result of the test, for versioning schema `version`:

`2.0.0` >
`1.128.8777` >
`1.128.8777-a` >
`1.128.8777-20` >
`1.128.8777-10` >
`1.128.8777-2` >
`1.128.8777-1.1` =
`1.128.8777-1.1+a` =
`1.128.8777-1.1+b` =
`1.128.8777-1.1+bb` >
`1.128.8777-1.0` >
`1.128.8777-1` >
`1.1` >
`1.0` >
`1` >
`0.0.0`

---

reverted the version format